### PR TITLE
feat(http): Throw errors for schema mismatch

### LINKED ISSUE #21357: Observe and fix HTTP response schemas
### Describe the proposed change(s).

The message `Response does not match schema` will be logged at the INFO level once per run.
We aim for eliminating them all before enforcing responses to strictly match provided schemas (throwing `ZodError` otherwise).

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Use strict parse function for schema
- Remove shallow copying

## Context

- Closes: #21357

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Hopefully should work well, but it would be great to observe errors/metrics after deploy, just in case

File: lib/util/http/index.ts [hunk 1]
```diff
@@ -259,19 +259,11 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
     const res = await this.request<ResT>(url, opts);
 
     if (!schema) {
-      return { ...res, body: res.body };
+      return res;
```
Comment (viceice) [inline lib/util/http/index.ts:262]: can we do these response refactoring first?

**Review by viceice — APPROVED** [review]
are we sure we've found Schema all mismatch before merging this?

**Comment by zharinov** [comment]
> are we sure we've found Schema all mismatch before merging this?

@rarkins Can we go ahead?

**Review by JamieMagee — APPROVED** [review]

**Comment by rarkins** [comment]
I didn't see any log messages in the hosted app, hopefully didn't miss them

**Pull request merged**